### PR TITLE
provisioner-cli: Update provisioner files to include profile id support

### PR DESCRIPTION
**Pull request by tiashah1**
Uncomment profile id steps in kustomize.yaml
Add extra validation for steps which require API key to be set

Fixes: #1243

### CODE REVIEW

**Review by jtumber-ibm — APPROVED** [review]
Tested with specifying just IAM_PROFILE_ID for an existing cluster using

```
IBMCLOUD_PROVIDER="ibmcloud"
IAM_PROFILE_ID="---"
PODVM_IMAGE_ID="---"
INSTANCE_PROFILE_NAME="bz2-2x8"
CAA_IMAGE_TAG="7322a2ff36d57feec125ea1a4aadde3714c842d8"
```

Worked as expected. Also checked the same worked for APIKEY as before. 

I haven't tried the provisioner step though

File: test/provisioner/provision_ibmcloud_initializer.go [hunk 1]
```diff
@@ -206,8 +208,14 @@ func initProperties(properties map[string]string) error {
 		}
 	}
 
-	if err := initVpcV1(); err != nil {
-		return err
+	if len(IBMCloudProps.ApiKey) > 0 {
+		if err := initClustersAPI(); err != nil {
+			return err
+		}
```
Comment (stevenhorsman) [inline test/provisioner/provision_ibmcloud_initializer.go:214]: Why was this moved out of the `if strings.EqualFold(needProvisionStr` block? Doesn't it mean that `TEST_PROVISION` will be ignored and the cluster will be created whenever the api key is set?

Comment (stevenhorsman) [reply test/provisioner/provision_ibmcloud_initializer.go:214]: So I think I've got this wrong after chatting to Jordan & Tia - this doesn't trigger the cluster creation, but just gets ready for it and it's needed for both provisioning and deprovisioning, so there is a change of behaviour, but it's because the deprovisioning would have failed previously which is a bit strange.

I think we basically want to get to the point that the provisioner-cli doesn't require any particular value of `TEST_PROVISION`, or `TEST_TEARDOWN` as we are explicitly calling those actions

File: test/provisioner/provision_ibmcloud_initializer.go [hunk 2]
```diff
@@ -204,10 +203,20 @@ func initProperties(properties map[string]string) error {
 		if len(IBMCloudProps.CosServiceURL) <= 0 {
 			return errors.New("COS_SERVICE_URL was not set, example: s3.us.cloud-object-storage.appdomain.cloud")
 		}
+	} else {
```
Comment (mattarnoatibm) [inline test/provisioner/provision_ibmcloud_initializer.go:206]: The `else` and `if` on the next line can be condensed into 1 line `else if`.

**Review by mattarnoatibm — CHANGES_REQUESTED** [review]
Thanks for this contribution. I've made 1 review comment.

File: test/provisioner/provision_ibmcloud_initializer.go (hunk 1 above)
Comment (stevenhorsman) [reply test/provisioner/provision_ibmcloud_initializer.go:215]: In chatting through with Tia, she has tested that the provision and deprovision actions work irrespective of the `TEST_PROVISION`, or `TEST_TEARDOWN` values. I think there is probably some clarification which could be done more generally to improve the interaction between the environment variables and actions, but that is the outside the scope of this PR, so this conservation can be resolved

File: test/provisioner/provision_ibmcloud_initializer.go (hunk 2 above)
Comment (tiashah1) [reply test/provisioner/provision_ibmcloud_initializer.go]: Thanks, I have just made that change

**Review by mattarnoatibm — APPROVED** [review]
Looks good to me (LGTM)

**Comment by stevenhorsman** [comment]
Can you `fixup`/`squash` the last three commits together please and update the commit message to reflect that it is fixing the `PODVM_IMAGE_ID` validation logic please? Thanks!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks @tiashah1!
